test/distributed/test_c10d.py

```diff
@@ -502,6 +507,22 @@ def test_nominal(self):
         # check with get
         self.assertEqual(b"value0", store0.get("key0"))
 
+    @retry_on_address_already_in_use_error
+    def test_tcp_store_timeout_set(self):
+        url = self.create_tcp_url()
+        test_store_timeout = timedelta(seconds=1)
+        gen0 = c10d.rendezvous(url + "&rank=0", timeout=test_store_timeout)
+        store0, rank0, size0 = next(gen0)
+        # this should time out in ~1s. If the timeout passed into rendezvous was
+        # not respected, it will take much longer to timeout.
+        start = time.time()
+        with self.assertRaisesRegex(RuntimeError, "Timeout"):
+            store0.get("nonexistant key")
+
+        end = time.time()
+        time_diff = end - start
+        self.assertGreater(test_store_timeout.seconds * 4, time_diff)
```

Comment:
The default timeout would've been 30 mins without this PR right? Lets just do * 10 to be conservative.